// @mohism/utils/dist/libs/logs/console-log.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConsoleLogger = void 0;
const colors_1 = require("colors");
const rightpad_1 = __importDefault(require("../rightpad"));
const dayjs_1 = __importDefault(require("dayjs"));
const stringify = (input) => {
    return (typeof input === 'string' ? input : JSON.stringify(input));
};
const LEVEL_MAP = {
    debug: 0,
    info: 1,
    warn: 2,
    error: 3,
    ultra: 99,
};
class ConsoleLogger {
    info(ctx) {
        if (process.env.LOG_LEVEL && LEVEL_MAP[process.env.LOG_LEVEL] > LEVEL_MAP.info) {
            return;
        }
        console.log(`${(0, colors_1.blue)((0, rightpad_1.default)('[INFO]', 6))} ${(0, colors_1.grey)((0, dayjs_1.default)().format('YYYY-MM-DD HH:mm:ss'))} ${(0, colors_1.grey)(stringify(ctx))}`);
    }
    warn(ctx) {
        if (process.env.LOG_LEVEL && LEVEL_MAP[process.env.LOG_LEVEL] > LEVEL_MAP.warn) {
            return;
        }
        console.log(`${(0, colors_1.yellow)((0, rightpad_1.default)('[WARN]', 6))} ${(0, colors_1.grey)((0, dayjs_1.default)().format('YYYY-MM-DD HH:mm:ss'))} ${(0, colors_1.grey)(stringify(ctx))}`);
    }
    // 为对下兼容而重复
    err(ctx) {
        if (process.env.LOG_LEVEL && LEVEL_MAP[process.env.LOG_LEVEL] > LEVEL_MAP.error) {
            return;
        }
        console.error(`${(0, colors_1.red)((0, rightpad_1.default)('[ERR]', 6))} ${(0, colors_1.grey)((0, dayjs_1.default)().format('YYYY-MM-DD HH:mm:ss'))} ${(0, colors_1.grey)(stringify(ctx))}`);
    }
    error(ctx) {
        if (process.env.LOG_LEVEL && LEVEL_MAP[process.env.LOG_LEVEL] > LEVEL_MAP.error) {
            return;
        }
        console.error(`${(0, colors_1.red)((0, rightpad_1.default)('[ERR]', 6))} ${(0, colors_1.grey)((0, dayjs_1.default)().format('YYYY-MM-DD HH:mm:ss'))} ${(0, colors_1.grey)(stringify(ctx))}`);
    }
}
exports.ConsoleLogger = ConsoleLogger;
exports.default = new ConsoleLogger();
