// @mohism/utils/dist/libs/logs/mohism-log.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MohismLogger = exports.LOG_DRIVER = void 0;
const log4js_1 = __importDefault(require("log4js"));
const os_1 = require("os");
const console_log_1 = require("./console-log");
var LOG_DRIVER;
(function (LOG_DRIVER) {
    LOG_DRIVER["console"] = "console";
    LOG_DRIVER["log4js"] = "log4js";
})(LOG_DRIVER = exports.LOG_DRIVER || (exports.LOG_DRIVER = {}));
class MohismLogger {
    constructor(conf) {
        this.conf = conf;
        if (conf.driver === LOG_DRIVER.log4js) {
            this.driver = this.initLog4js();
        }
        else {
            this.driver = this.initConsole();
        }
    }
    initLog4js() {
        const host = (0, os_1.hostname)();
        log4js_1.default.addLayout('json', () => {
            return (logEvent) => JSON.stringify({
                app: this.conf.appID,
                time: logEvent.startTime,
                level: logEvent.level.levelStr,
                data: logEvent.data[0],
                host,
            });
        });
        log4js_1.default.configure({
            appenders: {
                daily: {
                    type: 'dateFile',
                    layout: {
                        type: 'json'
                    },
                    filename: `${this.conf.logPath}/${this.conf.appID}`,
                    pattern: '.yyyy-MM-dd.log',
                    alwaysIncludePattern: true,
                    compress: true
                }
            },
            categories: {
                default: {
                    appenders: ['daily'],
                    level: 'info'
                }
            }
        });
        const logger = log4js_1.default.getLogger();
        if (this.conf.level) {
            logger.level = this.conf.level;
        }
        return logger;
    }
    initConsole() {
        return new console_log_1.ConsoleLogger();
    }
    info(ctx) {
        this.driver.info(ctx);
    }
    warn(ctx) {
        this.driver.warn(ctx);
    }
    err(ctx) {
        this.driver.error(ctx);
    }
    error(ctx) {
        this.driver.error(ctx);
    }
}
exports.MohismLogger = MohismLogger;
exports.default = MohismLogger;
