// @mohism/utils/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.reFlatting = exports.flatting = exports.toabcDef = exports.toAbcDef = exports.Maker = exports.Getter = exports.rightpad = exports.Logger = exports.exec = void 0;
var exec_1 = require("./libs/exec");
Object.defineProperty(exports, "exec", { enumerable: true, get: function () { return exec_1.exec; } });
var logs_1 = require("./libs/logs");
Object.defineProperty(exports, "Logger", { enumerable: true, get: function () { return __importDefault(logs_1).default; } });
var rightpad_1 = require("./libs/rightpad");
Object.defineProperty(exports, "rightpad", { enumerable: true, get: function () { return __importDefault(rightpad_1).default; } });
var getter_1 = require("./libs/getter");
Object.defineProperty(exports, "Getter", { enumerable: true, get: function () { return __importDefault(getter_1).default; } });
var lazy_1 = require("./libs/lazy");
Object.defineProperty(exports, "Maker", { enumerable: true, get: function () { return __importDefault(lazy_1).default; } });
var formatString_1 = require("./libs/formatString");
Object.defineProperty(exports, "toAbcDef", { enumerable: true, get: function () { return formatString_1.toAbcDef; } });
Object.defineProperty(exports, "toabcDef", { enumerable: true, get: function () { return formatString_1.toabcDef; } });
var flattingObject_1 = require("./libs/flattingObject");
Object.defineProperty(exports, "flatting", { enumerable: true, get: function () { return flattingObject_1.flatting; } });
Object.defineProperty(exports, "reFlatting", { enumerable: true, get: function () { return flattingObject_1.reFlatting; } });
__exportStar(require("./libs/array"), exports);
